lib/translate/inquiry.cc

```diff
@@ -0,0 +1,298 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "inquiry.h"
+#include "string.h"
+namespace inquiry {
+// Creates and validates a Inquiry Command struct
+std::optional<scsi_defs::InquiryCommand> raw_cmd_to_scsi_command(
```

Comment:
per https://github.com/googleinterns/scsi2nvme/pull/75#issuecomment-649020238 we are currently pivoting away from optionals